base/src/org/adempiere/process/GenerateNotRealizedGainLoss.java

```diff
@@ -0,0 +1,514 @@
+/**
+ * Copyright (C) 2003-2019, e-Evolution Consultants S.A. , http://www.e-evolution.com
+ * This program is free software, you can redistribute it and/or modify it
+ * under the terms version 2 of the GNU General Public License as published
+ * or (at your option) any later version.
+ * by the Free Software Foundation. This program is distributed in the hope
+ * that it will be useful, but WITHOUT ANY WARRANTY, without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along
+ * with this program, if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+ * For the text or an alternative of this public license, you may reach us
+ * or via info@adempiere.net or http://www.adempiere.net/license.html
+ * Email: victor.perez@e-evolution.com, http://www.e-evolution.com , http://github.com/e-Evolution
+ * Created by victor.perez@e-evolution.com , www.e-evolution.com
+ */
+
+package org.adempiere.process;
+
+import org.adempiere.exceptions.AdempiereException;
+import org.compiere.model.*;
+import org.compiere.util.DB;
+import org.compiere.util.Env;
+import org.compiere.util.KeyNamePair;
+import org.compiere.util.Msg;
+import org.compiere.util.Trx;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.reducing;
+import static java.util.stream.Collectors.toList;
+
+
+/**
+ * Generate Not Realized Gain / Loss for Multi-Currency Account
+ * This process takes the accounting records until a closing date for all multi-currency accounting accounts of the balance
+ * in the source currency of the transaction is converted to the currency of the accounting scheme at the exchange rate of the closing date,
+ * the difference with the current balance balance in currency records a journal for the difference as loss or gain in exchange
+ *
+ * @author victor.perez@e-evolution.com , www.e-evolution.com
+ */
+public class GenerateNotRealizedGainLoss extends GenerateNotRealizedGainLossAbstract {
+    private List<X_T_InvoiceGL> exchangeGainLossList = new ArrayList<>();
+    private HashMap<Integer, MJournalBatch> journalBatches = new HashMap();
+    private HashMap<String, MJournal> journals = new HashMap();
```

Comment:
Please remove unused declaration

Reply:
Change for it:
private HashMap<String, MJournal> journals = new HashMap<String, MJournal>();